solutions/cf-1555B.mdx

```diff
@@ -0,0 +1,341 @@
+---
+id: cf-1555B
+source: CF 1555B
+title: B. Two Tables
+author: David Zhang
+---
+
+<Info>
+Oftentimes, problems will try to trick you. For example, since the judge is asking for a double, you would assume you may need to some advanced geometry or some diagonal movements, but a good rule of thumb is to always stay simple. As the most elegant and more importantly the fastest solutions are the shortest.
+</Info>
+
+## Explanation
+
+To begin, you should never move the inital or first rectangle, in a diagonal motion. 
+If the first table were to move in a diagonal motion, the table could achieve the same result just moving in the x or y direction. 
+(The hypotenuse - the diagonal motion - will always be longer then the other sides - the x or y motion)
+
+Let's call: 
+* $x_{1}$ and $x_{2}$ the leftmost and rightmost positions of the inital rectangle respectively
+* $y_{1}$ and $y_{2}$ the bottommost and topmost positions of the inital rectangle respectively
+* $w_{1}$ and $h_{1}$ the width and height of the inital rectangle (which can be calculated by calulating the difference in x and y)
+* $w_{2}$ and $h_{2}$ the width and height of the second rectangle respectively
+* $W$ and $H$ the total width and height of the room
+
+More importantly, there are 4 different places the second rectangle could be placed: left, right, top, bottom.

```

Comment:
this should be `top left, top right, bottom left, bottom right` (?)

Reply:
I'll make it more clear. You can only place the second rectangle either directly to the right of it, left, top or bottom of the first rectangle.

Reply:
oops yeah, i completely forgot what this problem was about lol

Reply:
lol np